default/python/AI/AIstate.py

```diff
@@ -93,6 +94,37 @@ def __init__(self, aggression):
         self.empire_standard_enemy = (4, ((4, 1),), 0.0, 10.0)  # TODO: track on a per-empire basis
         self.empire_standard_enemy_rating = 40  # TODO: track on a per-empire basis
 
+        # Check the optionsDB for the behavior bypass values and create
+        # the character.
+        def get_behavior_bypass_value(name, default, sentinel):
+            """Fetch the bypassed behavior value or return the default"""
+
+            force_option = "AI.config.behavior.%s.force" % (name,)
+            if not fo.getOptionsDBOptionBool(force_option):
+                return default
+
+            per_id_option = "AI.config.behavior.%s.%s" % (name, fo.playerName())
+            all_id_option = "AI.config.behavior.%s.all" % (name,)
+
+            behavior = fo.getOptionsDBOptionInt(per_id_option)
+            if behavior is None or behavior == sentinel:
+                behavior = fo.getOptionsDBOptionInt(all_id_option)
+            # print 'NOT picv = ', per_id_option , ' bev = ', behavior
+            if behavior is None or behavior == sentinel:
+                print ("Bypass of %s behavior failed. Returning default value %s." % (name, str(default)))
+                print "To force %s behavior set %s to 1" % (name, force_option,)
+                print "Set %s to a value to force just %s's behavior." % (per_id_option, fo.playerName())
+                print "Set  %s to a value to force the behavior of all AIs not individually specified." % (all_id_option,)
+                behavior = default
```

Comment:
None of the cases are errors.  If the config value is missing then the dev (presumably) does not want to hard code the AI behavior in the config.xml file.  The print statements were only informative.  I moved the information into the docstring of the function.  Anyone hacking on the AI Character code will hopefully find it there.